src/Persistence/Sql.php

```diff
@@ -945,21 +952,34 @@ public function getFieldSqlExpression(Field $field, Expression $expression)
         return $expression->expr($mask, $prop);
     }
 
-    /**
-     * Last ID inserted.
-     *
-     * @return mixed
-     */
-    public function lastInsertId(Model $model)
+    private function getIdSequenceName(Model $model): ?string
     {
-        $seq = $model->sequence ?: null;
+        $sequenceName = $model->sequence ?: null;
 
-        // PostgreSQL PDO always requires sequence name in lastInsertId method as parameter
-        // So let's use its default one if no specific is set
-        if ($this->connection instanceof \atk4\dsql\Postgresql\Connection && $seq === null) {
-            $seq = $model->table . '_' . $model->id_field . '_seq';
+        if ($sequenceName === null) {
+            // PostgreSQL uses sequence internally for PK autoincrement,
+            // use default name if not set explicitly
+            if ($this->connection instanceof \atk4\dsql\Postgresql\Connection) {
+                $sequenceName = $model->table . '_' . $model->id_field . '_seq';
+            }
         }
 
-        return $this->connection->lastInsertId($seq);
+        return $sequenceName;
+    }
+
+    public function lastInsertId(Model $model): string
```

Comment:
Method description (comment) is lost :) and is it really always a string (not integer)?
Can't it also be `null` in case nothing is inserted yet?

Reply:
it must throw if nothing was inserted yet

Reply:
Is there a test for that?

Reply:
At least PostgreSQL will, this is quite vendor specific (some vendors may return 0), we should trust DSQL here as we use it here

Reply:
OK fine.